keps/sig-network/2091-cluster-network-policy/README.md

```diff
@@ -0,0 +1,1161 @@
+or modified by the admin.
+
+On the other hand, the `Allow` rules can be used to call out traffic in the cluster
+that needs to be allowed for certain components to work as expected (egress to
+CoreDNS for example). Those traffic should not be blocked when developers apply
+NetworkPolicy to their Namespaces which isolates the workloads.
+
+ClusterNetworkPolicy `Pass` rules allows an admin to delegate security posture for
+certain traffic to the Namespace owners. For example, allowing intra-tenant traffic
+can be delegated to tenant admins explicitly by the cluster admin with the use of `pass`
+rules.
+
+#### Priority
+
+The policy instances will be ordered based on the numeric priority assigned to the
+CNP. A lower number corresponds to a higher order. For example, CNP with `priority` set
+to "0" will be considered as the highest priority policy, while a large number, i.e.
+999, will be considered as low priority. Any numeric-valued priority, however, will have
+higher precedence over the namespaced NetworkPolicy instances in the cluster.
+
+Additionally, a special keyword `Baseline` can be used in the priority field to indicate
+that the rules in that policy instance shall be created at a priority lower than the Namespaced
+NetworkPolicies. Admission controllers will need to validate that only this string value or
+numeric values are set for the priority field.
+
+(TODO: double check this. There are benefits of having 0 be the lowest priority so that there is
+convenient to add higher and higher priority policies and critical override rules by the admin
+given any existing set of policies. It also enables an option of using priority 0 as a way to
+indicate the default network policy.)
+
+The precedence of the rules within a CNP will be determined by the order in which the rule
+is written. Thus, a rule that appears at the top of the ingress/egress rules would take the
+highest precedence.
+
+Conflict resolution: Two policies are considered to be conflicting if they are assigned the
+same `priority`. In order to avoid such conflicts, we propose to include validation for
+CNP resources, such that no two CNP instances can be created with the same `priority`.
+(TODO: Is such admission control viable?)
+
+For conformance and portability purposes, it will be required that implementations
+support up to 1000 CNP priorities (0 through 999) per cluster and up to 100 rules
+within a CNP instance.  An implementation can optionally choose to support larger numbers
+of priority values and rules, however if an implementation supports a
+smaller number of rules and instances than these recommended ranges, it will be considered
+non-conformant.  This ensures that policy manifests written using priorities 0 through
+999 for instance are portable across clusters and implementations without need to re-write
+policy manifests or re-adjust priority allocations when moving applications from 1
+kubernetes cluster to another.
```

Comment:
I don't think it makes sense to say that a larger range exists but only a subset is portable. If I want to create a "minimum-priority" policy, then to be portable I would have to make it be `Priority: 999`, but at that point it's no longer possible to use any `Priority` value larger than that.

You can say "implementations must support at least 1000 distinct `Priority` values" without also saying that those values have to be 0-999. Either that or we should just say that values outside the range 0-999 are not valid.

Reply:
Done